Implement PINT (Put IN one at a Time) CBA methodology

cdgaete opened the pull request:
Closes #136 #382

## Changes proposed in this Pull Request

Implements the PINT (Put IN one at a Time) methodology for Cost-Benefit Analysis as specified in TYNDP 2024 CBA guidelines.

**New scripts:**
- `scripts/cba/prepare_pint_reference.py`: Prepares the PINT reference network containing only `in_reference=True` projects
- `scripts/cba/prepare_pint_project.py`: Adds a single candidate project to the PINT reference network

**Modified files:**
- `rules/cba.smk`: Added `hurdle_costs` param to `prepare_pint_project` rule; updated `input_indicators()` to filter projects by CBA method
- `scripts/cba/_helpers.py`: Added `filter_projects_by_method()` with registry pattern for method-specific project filtering

## Tasks

- [x] Create PINT reference network script
- [x] Create PINT project network script
- [x] Add method-based project filtering (only `in_reference=False` for PINT)
- [x] Update workflow rules

## Workflow

To run PINT analysis, add `pint` to the `cba.methods` list in config:
```yaml
cba:
  methods:
  - toot
  - pint
```

## Open issues

None

## Notes

- PINT and TOOT use the same downstream rules (`solve_cba_network`, `make_indicators`, `collect_indicators`, `plot_indicators`)
- Project filtering by `in_reference` status happens at workflow level in `input_indicators()` for consistency
- Registry pattern (`METHOD_PROJECT_FILTERS`) allows easy addition of future CBA methods

## Checklist

- [x] I tested my contribution locally and it works as intended.
- [x] Code and workflow changes are sufficiently documented.
- [ ] Changed dependencies are added to `pixi.toml` (using `pixi add <dependency-name>`).
- [ ] Changes in configuration options are added in `config/config.default.yaml`.
- [ ] Changes in configuration options are documented in `doc/configtables/*.csv`.
- [ ] Changes in configuration options are added in `config/test/*.yaml`.
- [x] Open-TYNDP SPDX license header added to all touched files.
- [ ] Sources of newly added data are documented in `doc/data_sources.rst`.
- [ ] New rules are documented in the appropriate `doc/*.rst` files.
- [ ] A release note `doc/release_notes.rst` is added.
- [ ] Major features are documented with up-to-date information in `README` and `doc/index.rst`.
- [x] Module docstrings added to new Python scripts.

lisazeyen commented on scripts/cba/prepare_pint_reference.py:
```diff
@@ -0,0 +1,80 @@
+
+Projects with in_reference=False are candidates for PINT assessment and will
+be added one at a time in prepare_pint_project.py.
+"""
```
```suggestion

Projects with in_reference=True are candidates for TOOT assessment and will
be removed one at a time in prepare_toot_project.py.
"""
```

lisazeyen reviewed (changes requested):
@cdgaete PR looks mostly good. Since we know by now that there is only one CBA reference grid for PINT and TOOT, I would suggest renaming the `prepare_pint_reference` to `prepare_reference` and removing the `prepare_toot_reference`. There are some further changes needed than in the corresponding snakemake rule, see detailed comment below.

cdgaete commented on scripts/cba/_helpers.py:
```diff
@@ -4,6 +4,60 @@
+    return METHOD_PROJECT_FILTERS[method](projects, planning_horizon)
+
+
+def _filter_toot(projects: pd.DataFrame, planning_horizon: int) -> pd.DataFrame:
```
Yes, according to #365 discussion we will create a new issue and respective PR for this.

cdgaete commented:
## Update: Simplified PINT Implementation

The branch has been refactored to provide a minimal PINT implementation:

**Key changes:**
- PINT now shares the TOOT reference network (no separate `prepare_pint_reference`)
- Removed dependency on `method_assignment.csv` and related filtering logic
- `prepare_pint_project` takes the TOOT reference and adds one project

**Files changed:**
- `scripts/cba/prepare_pint_project.py` - adds single project to reference
- `rules/cba.smk` - simplified rules, PINT uses `prepare_toot_reference` output
- `config/*.yaml` - added `pint` to CBA methods

**Test results:** Both TOOT and PINT produce valid indicator results.

---

> ⚠️ **Follow-up PR needed:** Both TOOT and PINT currently assess **all** projects. A separate PR will add filtering based on `in_reference` column:
> - **TOOT**: projects where `in_reference=True`  
> - **PINT**: projects where `in_reference=False`

Ready for review.

lisazeyen commented on config/test/config.tyndp.yaml:
```diff
@@ -393,8 +393,9 @@ cba:
+  - pint
   projects:
-  - t1-t5
+  - t1-t35
```
revert to old selection of projects to not increase test run time
```suggestion
  - t1-t5
```

lisazeyen reviewed (approved):
Local test is running through, thanks @cdgaete !
